javadoc-crawler/src/main/java/io/opentelemetry/javadocs/JavaDocsCrawler.java

```diff
@@ -0,0 +1,175 @@
+    HttpClient client = HttpClient.newHttpClient();
+    List<Artifact> artifacts = getArtifacts(client);
+    if (artifacts.isEmpty()) {
+      logger.log(Level.SEVERE, "No artifacts found");
+      return;
+    }
+    logger.info(String.format(Locale.ROOT, "Found %d artifacts", artifacts.size()));
+
+    List<String> updated = crawlJavaDocs(client, artifacts);
+    if (updated.isEmpty()) {
+      logger.info("No updates were needed");
+      return;
+    }
+
+    logger.info("Artifacts that triggered updates:\n" + String.join("\n", updated));
+  }
+
+  static List<Artifact> getArtifacts(HttpClient client) throws IOException, InterruptedException {
+    int start = 0;
+    Integer numFound;
+    List<Artifact> result = new ArrayList<>();
+
+    do {
+      if (start != 0) {
+        Thread.sleep(THROTTLE_MS); // try not to DDoS the site, it gets knocked over easily
+      }
+
+      Map<?, ?> map = queryMavenCentral(client, start);
+
+      numFound =
+          Optional.ofNullable(map)
+              .map(mavenResult -> (Map<?, ?>) mavenResult.get("response"))
+              .map(response -> (Integer) response.get("numFound"))
+              .orElse(null);
+
+      List<Artifact> artifacts = convertToArtifacts(map);
+      result.addAll(artifacts);
+
+      start += PAGE_SIZE;
+    } while (numFound != null && start < numFound);
+
+    return result;
+  }
+
+  private static List<Artifact> convertToArtifacts(Map<?, ?> map) {
+    return Optional.ofNullable(map)
+        .map(mavenResults -> (Map<?, ?>) mavenResults.get("response"))
+        .map(response -> (List<?>) response.get("docs"))
+        .map(
+            docs -> {
+              List<Artifact> artifacts = new ArrayList<>();
+              for (Object doc : docs) {
+                Map<?, ?> docMap = (Map<?, ?>) doc;
+                String artifact = (String) docMap.get("a");
+                String version = (String) docMap.get("latestVersion");
+                if (artifact != null && version != null) {
+                  artifacts.add(new Artifact(artifact, version));
+                }
+              }
+              return artifacts;
+            })
+        .orElseGet(ArrayList::new);
+  }
+
+  private static Map<?, ?> queryMavenCentral(HttpClient client, int start)
+      throws IOException, InterruptedException {
+    URI uri =
+        URI.create(
+            String.format(
+                Locale.ROOT,
+                "%s%s&rows=%d&start=%d&wt=json",
+                MAVEN_CENTRAL_BASE_URL,
+                GROUP,
+                PAGE_SIZE,
+                start));
+
+    HttpRequest request = HttpRequest.newBuilder(uri).GET().build();
+
+    HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
+    if (response.statusCode() != 200) {
+      logger.log(
+          Level.SEVERE, "Unexpected response code: " + response.statusCode() + ": " + response);
+      throw new IOException("Unable to pull Maven central artifacts list");
```

Comment:
Might help to print the response status code and body (if available)

Reply:
do you mean adding it to the IOException message along with the log?

Reply:
Oh I missed that the logging. Between the logging and the IOException I think we're covered